src/WaveClip.cpp

```diff
@@ -1433,8 +1392,12 @@ void WaveClip::Paste(double t0, const WaveClip* other)
 /*! @excsafety{Strong} */
 void WaveClip::InsertSilence( double t, double len, double *pEnvelopeValue )
 {
-   sampleCount s0;
-   TimeToSamplesClip(t, &s0);
+   if (t == GetPlayStartTime() && t > GetSequenceStartTime())
+      ClearSequence(GetSequenceStartTime(), t);
+   else if (t == GetPlayEndTime() && t < GetSequenceEndTime())
+      ClearSequence(t, GetSequenceEndTime());
```

Comment:
I don't understand the four lines above.

This function isn't doing what the name suggests -- just inserting silence -- but it may also discard the trimmed samples.

Maybe there is a reason for that, can you explain?  Should there be a different function for that purpose?

But is it also a bug in that case, that the trim-left and trim-right values in the clip are not reset to 0?

Reply:
I think most paths to WaveClip::InsertSilence involve newly created empty tracks, so it doesn't matter, but an exception is that WaveTrack::Join calls WaveClip::AppendSilence which calls this.

Did you insert those lines only to handle this special case?

Reply:
The reason was that I considered `InserSilence(t0, length, envelope)` to be equivalent to `Paste(t0, silenceClip)`. And `Paste` discards the hidden part of the destination clip in case if the clip is prepended or appended. And I thought that this was the safest approach in these cases, but maybe you are right and hidden data should not be discarded (`Paste` should not discard too)  and `WaveTrack::Join` should handle this case instead.